app/models/generic_object.rb

```diff
@@ -0,0 +1,83 @@
+class GenericObject < ApplicationRecord
+  validates :name, :presence => true, :uniqueness => true
+
+  belongs_to :generic_object_definition
+  has_many   :custom_attributes, :as => :resource, :dependent => :destroy, :autosave => true
+
+  TYPE_MAP = {
+    :boolean  => ActiveRecord::Type::Boolean.new,
+    :datetime => ActiveRecord::Type::DateTime.new,
+    :time     => ActiveRecord::Type::Time.new,
+    :float    => ActiveRecord::Type::Float.new,
+    :integer  => ActiveRecord::Type::Integer.new,
+    :string   => ActiveRecord::Type::String.new,
+  }.freeze
+
+  def initialize(klass, attributes = {})
```

Comment:
I would really recommend against changing the iniatialize method on an ActiveRecord object...it can get very hairy.  Instead, expect that the caller will pass a separate attribute, perhaps `:generic_object_definition_name`, which you can then also make a getter/setter so it doesn't feel so foreign and so you don't have to override initialize (and you don't need that `definition_object` helper method).  So, something like...

```ruby
class GenericObject
  def generic_object_definition_name
    generic_object_definition.try(:name)
  end

  def generic_object_definition_name=(name)
    self.generic_object_definition = GenericObjectDefinition.find_by(:name => name)
  end
end

# Caller looks like:

go = GenericObject.create!(:generic_object_definition_name => "BlahBlah")
go.generic_object_definition
# => #<GenericObjectDefinition:0x007fa335336dc0 id: 123, name: "BlahBlah", ...>
go.generic_object_definition_name
# => "BlahBlah"
```